strict_build_flags=1 fails due to Wpedantic and __PRETTY_FUNCTION__

```
    galerautils/src/gu_log.h:73:33: warning: ISO C does not support '__PRETTY_FUNCTION__' predefined identifier [-Wpedantic]
            gu_log(level, __FILE__, __PRETTY_FUNCTION__, __LINE__,\
                                    ^
    galerautils/src/gu_log.h:85:23: note: in expansion of macro 'GU_LOG_C'
    #define gu_info(...)  GU_LOG_C(GU_LOG_INFO,  __VA_ARGS__, NULL)
```

This is with GCC 5.x. (Note, above it is a warning because strict_build_flags=0)

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,8 @@ addons:
   apt:
 #    sources:
 #      - ubuntu-toolchain-r-test
+#     https://github.com/travis-ci/travis-ci/issues/3668
+#     for alternatives in container builds
     packages:
       - libboost-program-options-dev
       - libssl-dev

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,13 +2,16 @@ language: cpp
 
 sudo: false
 
+cache:
+  - apt
+  - ccache
 compiler:
   - gcc
   - clang
 addons:
   apt:
-#    sources:
-#      - ubuntu-toolchain-r-test
+    sources:
+      - ubuntu-toolchain-r-test
 #     https://github.com/travis-ci/travis-ci/issues/3668
 #     for alternatives in container builds
     packages:
@@ -16,5 +19,15 @@ addons:
       - libssl-dev
       - check
       - scons
+      - gcc-4.9
+      - g++-4.9
 
-script: MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / }); MAX_JOBS=$(( ${RAM[1]} / 262144 )); echo Jobs\:\ $MAX_JOBS; ./scripts/build.sh -j $MAX_JOBS
+before_install:
+  - if [ ${CC} == "gcc" ]; then
+      export CC=`which gcc-4.9`;
+      export CXX=`which g++-4.9`;
+    fi
+
+script:
+  - ${CC} --version ; ${CXX} --version
+  - MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / }); MAX_JOBS=$(( ${RAM[1]} / 262144 )); echo Jobs\:\ $MAX_JOBS; ./scripts/build.sh -j $MAX_JOBS

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,13 +19,13 @@ addons:
       - libssl-dev
       - check
       - scons
-      - gcc-4.9
-      - g++-4.9
+      - gcc-5
+      - g++-5
 
 before_install:
   - if [ ${CC} == "gcc" ]; then
-      export CC=`which gcc-4.9`;
-      export CXX=`which g++-4.9`;
+      export CC=`which gcc-5`;
+      export CXX=`which g++-5`;
     fi
 
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,16 +1,18 @@
 language: cpp
 
+sudo: false
+
 compiler:
   - gcc
   - clang
+addons:
+  apt:
+#    sources:
+#      - ubuntu-toolchain-r-test
+    packages:
+      - libboost-program-options-dev
+      - libssl-dev
+      - check
+      - scons
 
-before_install:
-  # - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq || echo > /dev/null
-
-install:
-  - sudo apt-get install -y gcc g++ clang libboost-program-options-dev libssl-dev check scons
-  # - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 80 --slave /usr/bin/g++ g++ /usr/bin/g++-4.8
-  # - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.6 60 --slave /usr/bin/g++ g++ /usr/bin/g++-4.6
-
-script: MEM=$(head -1 /proc/meminfo); RAM=(${MEM// / }); MAX_JOBS=$(( ${RAM[1]} / 262144 )); echo Jobs\:\ $MAX_JOBS; ./scripts/build.sh -j $MAX_JOBS
+script: MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / }); MAX_JOBS=$(( ${RAM[1]} / 262144 )); echo Jobs\:\ $MAX_JOBS; ./scripts/build.sh -j $MAX_JOBS

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 language: cpp
 
-sudo: false
+sudo: true
 
 cache:
   - apt
@@ -27,9 +27,9 @@ addons:
 script:
   - if [ ${CC} != "clang" ]; then
       MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / });
-      MAX_JOBS=$(( ${RAM[1]} / 393216 )); echo Jobs\:\ $MAX_JOBS;
+      MAX_JOBS=$(( ${RAM[1]} / 393216 )); echo Max jobs\:\ $MAX_JOBS;
       CORES=$(grep -c ^processor /proc/cpuinfo);
-      [ $CORES -lt $MAX_JOBS ] && MAX_JOBS=$CORES;
+      [ $CORES -lt $MAX_JOBS ] && MAX_JOBS=$CORES; echo Real jobs\:\ $MAX_JOBS;
       ${CC} --version; ${CXX} --version;
       ./scripts/build.sh -j $MAX_JOBS;
       export CC=`which gcc-5`;

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 language: cpp
 
-sudo: true
+sudo: false
 
 cache:
   - apt

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,9 +5,11 @@ sudo: false
 cache:
   - apt
   - ccache
+
 compiler:
   - gcc
   - clang
+
 addons:
   apt:
     sources:
@@ -22,12 +24,19 @@ addons:
       - gcc-5
       - g++-5
 
-before_install:
-  - if [ ${CC} == "gcc" ]; then
+script:
+  - if [ ${CC} != "clang" ]; then
+      MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / });
+      MAX_JOBS=$(( ${RAM[1]} / 393216 )); echo Jobs\:\ $MAX_JOBS;
+      CORES=$(grep -c ^processor /proc/cpuinfo);
+      [ $CORES -lt $MAX_JOBS ] && MAX_JOBS=$CORES;
+      ${CC} --version; ${CXX} --version;
+      ./scripts/build.sh -j $MAX_JOBS;
       export CC=`which gcc-5`;
       export CXX=`which g++-5`;
+      ${CC} --version; ${CXX} --version;
+      ./scripts/build.sh -j $MAX_JOBS;
+    else
+      ${CC} --version ; ${CXX} --version;
+      ./scripts/build.sh;
     fi
-
-script:
-  - ${CC} --version ; ${CXX} --version
-  - MEM=$(head -n 1 /proc/meminfo); RAM=(${MEM// / }); MAX_JOBS=$(( ${RAM[1]} / 262144 )); echo Jobs\:\ $MAX_JOBS; ./scripts/build.sh -j $MAX_JOBS

--- a/galerautils/src/gu_log.h
+++ b/galerautils/src/gu_log.h
@@ -70,7 +70,7 @@ extern gu_log_severity_t gu_log_max_level;
 #if !defined(__cplusplus) || defined(GALERA_LOG_H_ENABLE_CXX)
 // NOTE: don't add "\n" here even if you really want to do it
 #define GU_LOG_C(level, ...)\
-        gu_log(level, __FILE__, __PRETTY_FUNCTION__, __LINE__,\
+        gu_log(level, __FILE__, __func__, __LINE__,\
                __VA_ARGS__, NULL)
 
 /**

